public/app/core/utils/explore.ts

```diff
@@ -218,6 +218,10 @@ export function parseUrlState(initial: string | undefined): ExploreUrlState {
     return errorResult;
   }
 
+  if (!config.featureToggles.legacyExploreCompactURL) {
+    return parsed;
```

Comment:
What happens if you try to access explore with a compact URL and the feature flag is false?

Reply:
it fallbacks to an empty state. what actually happens is that we use this in `ExplorePaneContainer` as 

```ts
const urlState = parseUrlState(props.urlQuery);
 // ...

  const { datasource, queries, range: urlRange, panelsState } = urlState || {};
```

given `urlState` is an array, `datasource`, `queries`, `range` and `panelsState` will just be undefined, like when you access explore without any URL param.

also, `isFromCompactUrl` (see following line in the same file) won't be defined, so the warning won't be shown (and the interaction not reported)

Reply:
I wonder if wed like to put an error message in the console or logs for this. With a message maybe like "A link to explore using a deprecated format was attempted, but there is a feature flag that disallows access. If access is critical, please add "legacyExploreCompactURL" to your config, but be warned we will be removing the feature flag and disabling access entirely at a later version. Please change the source of these URLs to prevent loss of functionality" and then maybe log it. I'm thinking for the use case for someone who has somehow missed the warnings prior to this and hits the limitation from the feature flag for the first time.

I'm not saying this is critical but wondering if some communication would be useful for this step.

Reply:
then i'd rather show the warning similarly to what we currently have, but with the content you proposed. console errors are pretty obscure to 90% of non-devs (and undiscoverable, one should know an error happened to look for a console message, and there will be no indication of that in the UI).

happy to do that if we want to surface it 👍

Reply:
updated to show an error alert instead in https://github.com/grafana/grafana/pull/60920/commits/e1a7691644da7d81a7b3bb9af741715fa24df191:

![Screenshot 2023-01-09 at 11 50 28](https://user-images.githubusercontent.com/1170767/211302181-699db486-34e8-4250-a1ef-7cbc765f306a.png)

with the feature flag on, the old message is shown:

![Screenshot 2023-01-09 at 11 51 54](https://user-images.githubusercontent.com/1170767/211302204-c2d03d26-8c56-415c-997b-7752d78dafcf.png)

i also slightly updated the styles so the alerts are correctly layed out.